<launch>

    <!-- color conversion from YUV422 to RGB -->
    <node pkg="ti_viz_nodes" type="viz_color_conversion" name="viz_color_conversion_node" output="screen" args="" required="true">
        <param name="width" value="1280"/>
        <param name="height" value="720"/>
        <!-- Input YUV image topic name to subscribe to -->
        <param name= "input_yuv_topic"    value = "camera/right/image_raw"/>
        <!-- Output RGB image topic name to publish to -->
        <param name= "output_rgb_topic"   value = "camera/right/image_raw_rgb"/>
        <!-- Input YUV image format, YUV420 or YUV422 -->
        <param name= "yuv_format"   value = "YUV422"/>
    </node>

    <!-- generate disparity map and confidence map from disparity_raw  -->
    <include file="$(find ti_viz_nodes)/launch/viz_disparity.launch"/>

    <!-- viz_semseg_cnn  -->
    <include file="$(find ti_viz_nodes)/launch/viz_semseg.launch"/>

    <!-- rviz -->
    <node pkg="rviz" type="rviz" name="rviz_sde" args="-d $(find ti_sde)/rviz/sde_semseg.rviz" />

</launch>
